### ISSUE TITLE
Compilation failed with CBLAS and LAPACK on ubuntu 20.06

### ISSUE THREAD

**Issue by GFuhr** [xianyi/OpenBLAS#2770]
Hello,

I tried to compile OpenBLAS with CBLAS, LAPACK, LAPACKE on Ubuntu 20.06 server and the make failed during tests.
The command I used :
'''
make NO_LAPACK=0 NO_CBLAS=0 NO_LAPACKE=0 NO_STATIC=0 NO_SHARED=0  -j8
'''
the tests failed at this point : 
'''
cc -O2 -DUTEST_CHECK -DSANITY_CHECK -DREFNAME=f_ -DMAX_STACK_ALLOC=2048 -fopenmp -Wall -m64 -DF_INTERFACE_GFORT -fPIC -DSMP_SERVER -DUSE_OPENMP -DNO_WARMUP -DMAX_CPU_NUMBER=48 -DMAX_PARALLEL_NUMBER=1 -DVERSION=\"0.3.10.dev\" -UASMNAME -UASMFNAME -UNAME -UCNAME -UCHAR_NAME -UCHAR_CNAME -DASMNAME= -DASMFNAME=_ -DNAME=_ -DCNAME= -DCHAR_NAME=\"_\" -DCHAR_CNAME=\"\" -DNO_AFFINITY -I..  -o openblas_utest utest_main.o test_min.o test_amax.o test_ismin.o test_rotmg.o test_axpy.o test_dotu.o test_dsdot.o test_swap.o test_rot.o test_potrs.o test_kernel_regress.o ../libopenblas_zenp-r0.3.10.dev.a -lm -lpthread -lgfortran -lm -lpthread -lgfortran -L/usr/lib/gcc/x86_64-linux-gnu/9 -L/usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu -L/usr/lib/gcc/x86_64-linux-gnu/9/../../../../lib -L/lib/x86_64-linux-gnu -L/lib/../lib -L/usr/lib/x86_64-linux-gnu -L/usr/lib/../lib -L/usr/lib/gcc/x86_64-linux-gnu/9/../../..  -lgfortran -lm -lgomp -lquadmath -lm -lpthread -lc
/usr/bin/ld: test_kernel_regress.o: in function `__ctest_kernel_regress_skx_avx_run':
test_kernel_regress.c:(.text+0xb4): undefined reference to `LAPACKE_dgesvd'
/usr/bin/ld: test_kernel_regress.c:(.text+0x13d): undefined reference to `cblas_dgemm'
/usr/bin/ld: test_kernel_regress.c:(.text+0x177): undefined reference to `cblas_dnrm2'
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:44: openblas_utest] Error 1
make[1]: Leaving directory '/opt/softs/sources/OpenBLAS/utest''''

If fails also without cblas and with or without shared libs
'''make NO_LAPACK=0 NO_CBLAS=1 NO_LAPACKE=0 NO_STATIC=0 NO_SHARED=0 -j8
 make NO_LAPACK=0 NO_CBLAS=1 NO_LAPACKE=0 NO_STATIC=0 NO_SHARED=1 -j8'''
a lot of undefined references to LAPACKE_* functions in linktest.c

the following make succeed (without cblas and without shared lib)
'''make NO_LAPACK=0 NO_CBLAS=1 NO_LAPACKE=0 NO_STATIC=0 NO_SHARED=1 -j8'''

I used gcc/gfortran v 9.3 and I didn't install lapack/blas via "apt"
Edit :
I tried with the latest archive and the from the develop git branch (August 8, 2020) but issue is the same in both cases

G.

**Comment by martin-frbg** [comment]
At least with earlier versions, it is probably safer to just leave out all the NO_something=0 - I think I got all the places fixed where the code just checked if an option was defined at all, but maybe some are still lurking. Any error messages earlier in the build, where the LAPACKE functions are supposed to be compiled ? Also, did you run `make clean` between build attempts with different options ?

**Comment by GFuhr** [comment]
yes I did the "make clean" everytime. I checked on both a real ubuntu 20.06server and on a virtualbox 20.06 ubuntu server and results are identical.
If I run only :
'''
make NO_STATIC=0 NO_SHARED=0 ONLY_CBLAS=0 BINARY=64 -j8
'''
everything seems to work well.

From the previous make command, I didn't grep any other error before the mentionned one.
From the failed builed, if I check the generated lib : 
'''
nm ./libopenblas.a | grep cblas_
'''
nothing is returned
